Fix: Update Finder Configuration to Follow Symlinks

SebBareyre opened the pull request:
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | This pull request addresses an issue where symlinks were not being followed in the file system. This is critical for cases where the application relies on symlinked directories. Modified the Finder configuration to follow symlinks by adding the followLinks() method. Updated relevant tests to account for the changes.
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | 
| Sponsor company   | ohmyweb
| How to test?      | https://friends-of-presta.slack.com/archives/CKJS3QZHR/p1688564319556699?thread_ts=1688563573.279279&cid=CKJS3QZHR

FabienPapet reviewed (approved):
Nice catch

jolelievre reviewed (approved):
Thanks @SebBareyre

matks reviewed (dismissed):
Hi @SebBareyre I attempted to reproduce your issue but it works very well on my localhost.

I moved a module folder outside of my prestashop folder and symlinked it: it works.
I moved classes/order outside of my prestashop folder and symlinked it: it works.

Could you narrow down what configuration is needed for symlinks to fail? If I can't reproduce, I can't verify.

It is possible the problem is only a configuration issue on your side

I don't think adding `followLinks()` is bad so I guess it cannot hurt but I wish to be sure :wink:

matks commented:
(I spent almost one hour and half tinkering with prestashop and its filesystem to "make it fail" to reproduce the problem and could not, no matter what. PrestaShop's quite robust actually 😁 )

SebBareyre commented:
> Hi @SebBareyre I attempted to reproduce your issue but it works very well on my localhost.
> 
> I moved a module folder outside of my prestashop folder and symlinked it: it works. I moved classes/order outside of my prestashop folder and symlinked it: it works.
> 
> Could you narrow down what configuration is needed for symlinks to fail? If I can't reproduce, I can't verify.
> 
> It is possible the problem is only a configuration issue on your side
> 
> I don't think adding `followLinks()` is bad so I guess it cannot hurt but I wish to be sure 😉

Hi @matks 
The problem appear on my side on the overrides folder. Try moving the override/classes and override/controllers folders outside of your PrestaShop folder and symlink it. I confirm that according to the doc of the Finder component of symfony. you have to add a `followLinks() `so that it can access it.

matks commented:
I'm sorry to disappoint you but... it works well on my localhost, without `followLinks()` 😅 

It's possible some filesystem natively follow symlinks while others do not. Also the webserver configuration could have an impact: nginx and apache for example do handle symlinks differently.
